docs(grid): expands design guidelines for grid layout. (#3672)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/grid.md

id: Grid
section: layouts
---
The **grid** layout places content on a fixed 12 column grid.

## Elements

![Example of grid layout.](./img/grid-example.png)
1. **Grid**: The grid container, which holds all grid items.
1. **Grid item**: An individual section of content to be placed in the grid.

## Usage

> Grid items can be ordered in a number of ways to create groups. The span value of all grid items must equal 12, but this can be achieved by combining different section sizes.

Author: ```suggestion
Grid items can be ordered in a number of ways to create groups. The span value of all grid items must equal 12, but this can be achieved by modifying the span of each item to create different combinations of section sizes:
```

> However, by following the rule that all items should span up to 12, the grid can easily be modified and readjusted:

Reviewer: this is small but for some reason the "however" feels wrong here, cause you already have the "but this can be achieved..." which already has the positive spin to having to equal 12.

Author: that's a good catch - I agree! I updated this phrasing, lmk what you think

> However, by following the rule that all items should span up to 12, the grid can easily be modified and readjusted:

Author: ```suggestion
```

id: Grid
section: layouts
---

> The **grid** layout places content on a fixed-size 12 column grid.

Author: ```suggestion
The **grid** layout places content on a fixed 12 column grid.
```

## Elements

![Grid example image](./img/grid-example.png)
1. **Grid**: The grid container, which holds all grid items.
1. **Grid item**: An individual section of content to be placed in the grid.

## Usage

> Grid items can be ordered in a number of ways to create groups. The span value of all grid items must equal 12, but this can be acheived through any combination of section sizes.

Author: ```suggestion
Grid items can be ordered in a number of ways to create groups. The span value of all grid items must equal 12, but this can be achieved by combining different section sizes.
```

For example, our [basic React example](/layouts/grid) organizes the following grid:

![Grid from React example](./img/original-grid-span.png)

> However, by following the rule that all items should span up to 12, the grid can easily be modified and readjusted:

Author: ```suggestion
However, by following the rule that section sizes can be customized as long as the span equals 12, the grid can easily be modified and readjusted:
```

![Adjusted React example grid](./img/adjusted-grid-span.png)

## Variations

### Gutters around items

To adjust spacing between items, you can apply gutters.

![Grid with gutter](./img/grid-gutters.png)

### Adjusted for different screens 

By adjusting the column span, row span, and order at various [breakpoints](/developer-resources/global-css-variables#breakpoint-variables-and-class-suffixes), you can adapt the grid for different screen sizes.

> [A couple of image examples?]

Reviewer: As a designer, yes I would add image examples to understand. But as a dev, maybe linking to those variables is enough? I'll let Lucia opine haha

Author: I opted to remove that section, because I don't know how to visualize that info, but lmk if you think it's important to re-add!
